### ISSUE TITLE
intersect can also support fractions easily

### ISSUE THREAD

**Issue by trent2** [josdejong/mathjs#2291]
All computation in intersect also work with fractions since they only use basic arithmetic commands. I tried to enable this by simply putting

```javascript
import { isBigNumber, isFraction } from '../../utils/is.js';
```
at the top and inserting ``|| isFraction(a)`` further down:
```javascript
  function _isNumeric(a) {
    // intersect supports numbers and bignumbers ((and maybe fractions!?))
    return typeof a === 'number' || isBigNumber(a) || isFraction(a);
  }
```
in ``intersect.js``. This works with all desired cases, e. g.
```javascript
intersect([1,-1,1], [2,0,2], [4, 0, 1], [2, 2, 5])
intersect([1,-1,1], [2,0,2], [4, 0, 1], [fraction(2,1), 2, 5])
```
Both return the correct result ``[3,1,3]``; the second call returns an array of fractions ``[3/1, 1/1, 3/1]`` (which is ok imho).

I don't really know what cases have to be tested so I thought I'd mention this as an easily implementable feature request, rather than providing a pull request.

**Comment by m93a** [comment]
Hey Robert,
thanks for looking into this. Supporting fractions make sense to me.

This would make for a nice PR if anybody feels like it :)

**Comment by thetazero** [comment]
#2329 Tried to implement this.

**Comment by josdejong** [comment]
The fix of @thetazero is published now in `v10.0.2`.
